ksqldb-functional-tests/src/test/resources/query-validation-tests/row_offset_and_partition.json

```
@@ -0,0 +1,43 @@
+{
+  "comments": [
+    "Tests ensuring ROWPARTITON and ROWOFFSET work as user column names before KLIP-50 is integrated"
+  ],
+  "tests": [
+    {
+      "name": "create ROWOFFSET and ROWPARTITION user columns as STRINGs",
+      "statements": [
+        "CREATE STREAM INPUT (id int KEY, val STRING, other_val STRING) WITH (kafka_topic='test_topic', value_format='JSON');",
+        "CREATE STREAM OUTPUT AS SELECT id, val as ROWPARTITION, other_val as ROWOFFSET FROM INPUT;"
+      ],
+      "inputs": [
+        {"topic":  "test_topic", "key": 1, "value": {"val": "a", "other_val": "b"}}
+      ],
+      "outputs": [
+        {"topic":  "OUTPUT", "key": 1, "value": {"ROWPARTITION": "a", "ROWOFFSET":  "b"}}
+      ],
+      "post": {
+        "sources": [
+          {"name": "OUTPUT", "type": "stream", "schema": "ID INT KEY, ROWPARTITION STRING, ROWOFFSET STRING"}
+        ]
+      }
+    },
+    {
+      "name": "test ROWOFFSET and ROWPARTITION user columns with data types the pseudocolumns will later represent",
```

We don't want to have to change the test name once KLIP-50 is implemented (we'd have to remember to rename the historic plan files if so).
```suggestion
      "name": "test ROWOFFSET and ROWPARTITION user columns with pseudocolumn data types",
```

---

Woops, I totally missed the source schema thing the first time around. Hopefully the latest commits address everything.